spring-cloud-dataflow-container-registry/src/main/java/org/springframework/cloud/dataflow/container/registry/authorization/DropAuthorizationHeaderRequestRedirectStrategy.java

```diff
@@ -92,109 +91,75 @@ public URI getLocationURI(final HttpRequest request, final HttpResponse response
 		URI httpUriRequest = super.getLocationURI(request, response, context);
 		String query = httpUriRequest.getQuery();
 		String method = request.getMethod();
-
 		// Handle Amazon requests
 		if (StringUtils.hasText(query) && query.contains(AMZ_CREDENTIAL)) {
 			if (isHeadOrGetMethod(method)) {
-                try {
-                    return new DropAuthorizationHeaderHttpRequestBase(httpUriRequest, method).getUri();
-                } catch (URISyntaxException e) {
-                    throw new HttpException("Unable to get location URI", e);
-                }
-            }
+				removeAuthorizationHeader(request, false);
+				try {
+					if (isHeadMethod(method)) {
+						return new HttpHead(httpUriRequest).getUri();
+					}
+					else {
+						return new HttpGet(httpUriRequest).getUri();
+					}
+				}
+				catch (URISyntaxException e) {
+					throw new HttpException("Unable to get location URI", e);
+				}
+			}
```

Comment:
These lines are repeated in all 3 cases. Let's factor this out for DRY.

Reply:
Done